Sources/MeiliSearch/Model/Task.swift

```diff
@@ -21,7 +21,20 @@ public struct Task: Codable, Equatable {
   public let status: Task.Status
 
   /// Type of the task.
-  public let type: String
+  //public let type: String
+    public enum Type:Codable {
+        case indexCreation
+        case indexUpdate
+        case indexDeletion
+        case indexSwap
+        case documentAdditionOrUpdate
+        case documentDeletion
+        case settingsUpdate
+        case dumpCreation
+        case taskCancelation
+        case taskDeletion
+        case snapshotCreation
```

Comment:
You still have to fix these typing errors to make your tests pass:

<img width="528" alt="image" src="https://user-images.githubusercontent.com/4116980/227534860-3161714a-2073-4e0e-9df9-e507d968c69b.png">

Example, from this:  

`XCTAssertEqual("indexCreation", task.type)` 
to this:
`XCTAssertEqual(Task.TaskType.indexCreation, task.type)`